src/core/Akka.Streams.Tests/Dsl/AsyncEnumerableSpec.cs

```diff
@@ -230,6 +225,47 @@ public async Task AsyncEnumerableSource_Must_Cancel_Running_Source_If_Downstream
             // The cancellation token inside the IAsyncEnumerable should be cancelled
             await WithinAsync(3.Seconds(), async () => latch.Value);
         }
+        
+        /// <summary>
+        /// Reproduction for https://github.com/akkadotnet/akka.net/issues/6280
+        /// </summary>
+        [Fact]
+        public async Task AsyncEnumerableSource_BugFix6280()
+        {
+            async IAsyncEnumerable<int> GenerateInts()
+            {
+                foreach (var i in Enumerable.Range(0, 100))
+                {
+                    if (i > 50)
+                        await Task.Delay(1000);
+                    yield return i;
+                }
+            }
+
+            var source = Source.From(GenerateInts);
+            var subscriber = this.CreateManualSubscriberProbe<int>();
+
+            await EventFilter.Warning().ExpectAsync(0, async () =>
+            {
+                var mat = source
+                    .WatchTermination(Keep.Right)
+                    .ToMaterialized(Sink.FromSubscriber(subscriber), Keep.Left);
+            
+#pragma warning disable CS4014
+                var task = mat.Run(Materializer);
+#pragma warning restore CS4014
+
+                var subscription = subscriber.ExpectSubscription();
+                subscription.Request(50);
+                subscriber.ExpectNextN(Enumerable.Range(0, 50));
+                subscription.Request(10); // the iterator is going to start delaying 1000ms per item here
```

Comment:
Force the `IAsyncEnumerable` to emit the elements that start producing 1s delays.